srcpkgs/systemd-boot/files/kernel.d/systemd-boot.post-install

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+#
+# Kernel hook for systemd-boot.
+#
+# Arguments passed to this script: $1 pkgname, $2 version.
+#
+PKGNAME="$1"
+VERSION="$2"
+
+. "$ROOTDIR/etc/default/systemd-boot"
+
+if [ "$SYSTEMD_BOOT_DISABLE" ]; then
+	exit 0
+fi
+
+is_installed="$(bootctl is-installed)"
+
+if [ "$is_installed" != "yes" ]; then
+	echo "systemd-boot is not installed to the ESP"
+	echo "run 'bootctl install' to install it"
+	exit 0
+fi
+
+BOOT="${BOOT:-/boot}"
+ESP="${ESP:-"$BOOT"}"
+
+BOOT="$ROOTDIR/$BOOT"
+ESP="$ROOTDIR/$ESP"
+
+[ -d "$ESP" ] || exit 0
+
+ARCH="$(xbps-uhelper arch)"
+
+# EFI_ARCH is the identifier from the UEFI specification
+case "$ARCH" in
+	x86_64*) EFI_ARCH="x64";;
+	i686*) EFI_ARCH="IA32";;
+	aarch64*) EFI_ARCH="AA64";;
+esac
+
+entries="$ESP/loader/entries"
+name="void-$ARCH-$VERSION"
+entry="$entries/$name.conf"
+
+# check if user provided options in /etc/default/systemd-boot
+if [ -z "$CMDLINE" ]; then
+	# get UUID from fstab
+	uuid=$(awk '$2 == "/" { print $1 }' < /etc/fstab | sed 's/^UUID=//')
+	CMDLINE="root=UUID=$uuid quiet rw"
```

Comment:
```suggestion
	rdev="$(awk '$2 == "/" { print $1 }' < /etc/fstab)" || rdev=
	if [ -z "${rdev}" ]; then
		echo "failed to determine root device, aborting"
		exit 1
	fi
	CMDLINE="root=UUID=$uuid quiet rw"
```
Not expecting that `fstab` uses a `UUID` designation is a bit more robust, although it still doesn't consider situations like

    # / some valid comment that will fool the parser

Might be worth doing a bit more validation on fields 1 and 2 to make sure they really look like a root entry.

Reply:
> what about leading space situations?

I'm not sure what you mean. Could you provide an example?

Reply:
```
     # comment with leading whitespace
```

Reply:
fixed the situation where there is any number of spaces preceding a `#`